### PR TITLE
Precise lerp

### PR DESCRIPTION
**Pull request by ravo10**
Could add better performance in milliseconds.

### CODE REVIEW

File: garrysmod/lua/includes/util.lua [hunk 1]
```diff
@@ -1,460 +1,469 @@
...
+	if ( delta < 0 ) then return from end
+
+	return from + ( to - from ) * delta
+
+end
+
+--[[---------------------------------------------------------
+	Precise lerp - Basically the same as `Lerp( ... )`, but will always guarantee that the return value = `to` when `delta` = 1.
+-----------------------------------------------------------]]
+function LerpPrecise( delta, from, to )
```
Comment (ravo10) [inline garrysmod/lua/includes/util.lua:251]: This is the change!

File: garrysmod/lua/includes/util.lua [hunk 2]
```diff
@@ -1,460 +1,469 @@
...
+
+	return (1 - delta) * from + delta * to;
+
+end
```
Comment (ravo10) [inline garrysmod/lua/includes/util.lua:255]: This is the change!

**Comment by Donkie** [comment]
This will not get merged until you properly fix the line endings or whatever is causing all the changes

**Comment by ravo10** [comment]
> This will not get merged until you properly fix the line endings or whatever is causing all the changes

I have no idea why that happend... I only staged the lines I wanted. I have to look into it.. I have not used git allot before.

**Comment by ravo10** [comment]
OK fixed!!

**Comment by ravo10** [comment]
> OK fixed!!

@neico @Donkie I used `git add --renormalize -p` and the `git rebase -i HEAD~[n]` and that fixed it for me. Thanks for the help.

**Comment by Donkie** [comment]
Why is this needed? Your code should rarely if ever depend on a float number being an exact value.

**Comment by thegrb93** [comment]
I have seen this issue in the professional world so there are cases. Personally I'd prefer this replace the original.

**Comment by Kefta** [comment]
The original should just be moved to Lua.

**Comment by robotboy655** [comment]
The original is in Lua.

**Comment by Kefta** [comment]
I'm blind btw. Sorry it had to come out like this.

**Comment by robotboy655** [comment]
Surely we can just change the existing `Lerp` to do this and not add a duplicate function?

**Pull request closed without merging**